cdp_scrapers/scraper_utils.py

```diff
@@ -66,6 +78,263 @@ def str_simplified(input_str: str) -> str:
+
+    Notes
+    -----
+    Remove roles[#] if roles[#].body in static_data.primary_bodies.
+    Use known_static_data.persons[#].seat.roles instead.
+
+    If roles[i].body not in static_data.primary_bodies,
+    roles[i].title cannot be "Councilmember" or "Council President".
+
+    Use "City Council" and "Council Briefing"
+    if known_static_data.primary_bodies is empty.
+    """
+    if roles is None:
+        roles = []
+
+    if not known_static_data or not known_static_data.primary_bodies:
+        # Primary/full council not defined in static data file
+        primary_body_names = ["city council", "city briefing"]
+    else:
+        primary_body_names = [
+            body_name.lower() for body_name in known_static_data.primary_bodies.keys()
+        ]
+
+    try:
+        have_primary_roles = len(known_static_data.persons[person_name].seat.roles) > 0
+    except (KeyError, AttributeError, TypeError):
+        have_primary_roles = False
+
+    class CouncilMemberTerm(NamedTuple):
+        start_datetime: datetime
+        end_datetime: datetime
+        roles_index: int
+
+    terms: List[CouncilMemberTerm] = []
+
+    for i, role in enumerate(roles):
+        role_title = str_simplified(role.title).lower()
+
+        if (
+            role.body is not None
+            and role.body.name is not None
+            and str_simplified(role.body.name).lower() in primary_body_names
+        ):
+            if have_primary_roles:
+                # Primary roles for council defined in static data file.
+                # Use static information and ignore this scraped info.
+                roles[i] = None
+            elif re.search("|".join(president_patterns), role_title, re.I) is not None:
+                role.title = RoleTitle.COUNCILPRESIDENT
+            else:
+                role.title = RoleTitle.COUNCILMEMBER
+                # get all councilmember terms
+                if role.start_datetime is not None and role.end_datetime is not None:
+                    terms.append(
+                        CouncilMemberTerm(role.start_datetime, role.end_datetime, i)
+                    )
+        # Role is not for a primary/full council
+        # Role.title cannot be Councilmember or Council President
+        elif "vice" in role_title:
+            role.title = RoleTitle.VICE_CHAIR
+        elif re.search("|".join(president_patterns), role_title, re.I) is not None:
+            role.title = RoleTitle.CHAIR
+        elif "alt" in role_title:
+            role.title = RoleTitle.ALTERNATE
+        else:
+            role.title = RoleTitle.MEMBER
+
+    if have_primary_roles:
+        # don't forget to include info from the static data file
+        roles.extend(known_static_data.persons[person_name].seat.roles)
+    if len(terms) == 0:
+        # no Councilmember roles dynamically scraped
+        return reduced_list(roles)
+
+    # sort in asc order of start_datetime and end_datetime.
+    terms.sort()
```

Comment:
Yes the latter, since

```Python
list_of_tuple = [(3, 4), (1, 2)]
list_of_tuple.sort()
print(list_of_tuple)
# [(1, 2), (3, 4)]
```

But I see your point, though. Would the first way be more readable/understandable code?

Reply:
The first way is easier to read for me. The second way requires knowing CouncilMemberTerm is a tuple and its internal order. The first way would reduce cognitive load while reading the code. Plus, you wouldn't need the comment anymore.